SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -271,6 +273,13 @@ + (void)setupWithSafeBlockingMigrations:(void (^_Nonnull)(void))safeBlockingMigr
     // app to crash on launch.
     safeBlockingMigrationsBlock();
 
+    // We need to do this _before_ anyone accesses the session or identity store state.
+    //
+    // None of this state is used by a view, so we can safely do this before the
```

Comment:
>  // None of this state is used by a view, so we can safely do this before the

We're talking about **YapDB** views here (as opposed to UIView's), right? 

Regardless of that fact that **we** know that no YapDB views use these collections, the views themselves don't know what happened in the write transaction (they only know *that* a write transaction occurred), so creating a write transaction before a view is registered will cause that view to be re-built from scratch, as soon as it does register, right?

Reply:
Huh, you might be right.  I guess we should use _read_ transactions when migrating/copying collections.